src/main/java/net/minecraftforge/common/config/ConfigManager.java

```diff
@@ -261,6 +263,8 @@ private static void sync(Configuration cfg, Class<?> cls, String modid, String c
                         {
                             Property property = property(cfg, wrapper.getCategory(), suffix, propType, adapt.isArrayAdapter());
 
+                            property.setHasSlidingControl(hasSlidingControl);
```

Comment:
Setting up field properties like requiresMcRestart, or here, hasSlidingControl, is handled via IFieldWrapper.setupConfiguration(). This generalizes the setup.
Please consider moving it to that method.